### ISSUE TITLE
YAM crash if the connection fails.

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#201]
**Originally by _opionline@yahoo.de_ on 2010-09-01 14:32:05 +0200**

___
Sometimes a connection to a yahoo server can't established.
Today was such case and YAM produce some MuTools hits.

So i do a @all all log and redirect the MuTools Output to
the same window. I attach the interest part as sashimi.out.

The debug log for this session:
```
SERVER[0060]: +OK hello from popgate 2.46.7 on pop015.mail.ukl.yahoo.com
CLIENT[0016]: USER xxxxxxxxx
SERVER[0024]: +OK password required.
CLIENT[0015]: PASS xxxxxxxx
SERVER[0066]: -ERR [AUTH] Error logging in. Please visit http://mail.yahoo.com
CLIENT[0006]: QUIT
SERVER[-001]: 
```

If a connection can established then i get no hits.

Used Version: YAM 2.7-dev [OS3/m68k] (01.09.2010)

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2010-09-01 14:32:16 +0200**

___
**Attachment added:** [`sashimi.out`](https://yam.ch/attachments/201/sashimi.out) (10.8 KiB)

**Comment by tboeckel** [comment]
**Originally on 2010-09-01 15:22:48 +0200**

___
In (aa017c5):
 * tcp/pop3.c: added a NULL pointer check to catch the situation that a failed
  connection is attempted to be shut down. This closes #201.
